Support compute_uinit in coreml backend when calling ct.convert

seayoung1112 opened the pull request:
Summary:
It seems currently the compute unit value set in executorch/examples/models/llama2/lib/partitioner_lib.py?lines=74 is not used during the convert, adding the support in this change.

However to my surprise, the model name in the actual ios instrument result has the right suffix (see the image below), I couldn't find anywhere in the code how this is set and I have no way to validate if it's just part of the name or it's indeed set correctly through another path. I'll abandon this diff if it's not needed actually...

 {F1718372833}

Differential Revision: D58981934

YifanShenSZ commented:
I think `executorch/examples/models/llama2/lib/partitioner_lib.py?lines=74` is used. Quoting reply from a [relevant PR](https://github.com/pytorch/executorch/pull/3683)
> using `CPU_AND_GPU` was intentional, we are investigating why the model falls off ANE and once that's resolved we can set it to `ALL` .

YifanShenSZ reviewed (approved):
It is still good to expose compute unit control in general cases, though, if you find it necessary to use not only the default compute unit `ALL`

Commit:
Support compute_uinit in coreml backend when calling ct.convert (#4059)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4059

It seems currently the compute unit value set in https://www.internalfb.com/code/fbsource/fbcode/executorch/examples/models/llama2/lib/partitioner_lib.py?lines=74 is not used during the convert, adding the support in this diff.

However to my surprise, the model name in the actual ios instrument result has the right suffix (see the image below), I couldn't find anywhere in the code how this is set and I have no way to validate if it's just part of the name or it's indeed set correctly through another path. I'll abandon this diff if it's not needed actually...

 {F1718372833}

Reviewed By: cccclai

Differential Revision: D58981934
